library/src/scala/compiletime/package.scala

```diff
@@ -95,28 +95,16 @@ package object compiletime {
     // implemented in dotty.tools.dotc.typer.Inliner
     error("Compiler bug: `constValue` was not evaluated by the compiler")
 
-  /**
-   * Use this type to widen a self-type to a tuple. E.g.
-   * ```
-   * val x: (1, 3) = (1, 3)
-   * val y: Widen[x.type] = x
-   * ```
-   */
-  type Widen[Tup <: Tuple] <: Tuple = Tup match {
-    case EmptyTuple => EmptyTuple
-    case h *: t => h *: t
-  }
-
   /** Given a tuple type `(X1, ..., Xn)`, returns a tuple value
    *  `(constValue[X1], ..., constValue[Xn])`.
    */
-  inline def constValueTuple[T <: Tuple]: Widen[T]=
+  inline def constValueTuple[T <: Tuple]: T =
     val res =
       inline erasedValue[T] match
         case _: EmptyTuple => EmptyTuple
         case _: (t *: ts) => constValue[t] *: constValueTuple[ts]
       end match
-    res.asInstanceOf[Widen[T]]
+    res.asInstanceOf[T]
```

Comment:
Is that cast actually still needed?

Reply:
Yes, before inlining the type of `res` is `Tuple`.

Reply:
Can these two be implemented in terms of `map`, which already deals with casting it.

Reply:
The issue is that we do not have a tuple on which we could call `map`.